packages/portal-proto/src/components/Modals/AgreementModal.tsx

```diff
@@ -20,6 +20,13 @@ export const AgreementModal = ({
 }): JSX.Element => {
   const dispatch = useCoreDispatch();
   const [checked, setChecked] = useState(false);
+
+  useEffect(() => {
+    if (!openModal) {
+      setChecked(false);
+    }
+  }, [openModal]);
```

Comment:
having to do this because of how we are rendering our modals. They are not un-mounted until there parent component goes away.
Could've added it onClose but `DownloadButton` acts a bit different so I added an effect.

Reply:
Not of big fan of it but it's what it is ATM.

Reply:
I'm okay with this. Once we move to mantine v7 can either investigate using: https://mantine.dev/x/modals/ or write our own V2Modals Manager